Introduce APIRule v1beta2 (#996)
kyma-project/api-gateway: docs/user/custom-resources/apirule/04-60-apirule-v1beta2-custom-resource.md

Reviewer: Should we also add some information about depreaction in the [v1beta1 documentation](https://github.com/kyma-project/api-gateway/blob/main/docs/user/custom-resources/apirule/04-10-apirule-custom-resource.md)?

Reviewer 2: @werdes72 please take on this.

> # APIRule v1beta2 Custom Resource <!-- {docsify-ignore-all} -->

Reviewer 3: ```suggestion
# APIRule v1beta2 Custom Resource
```

| **rules.service**                                |  **NO**   | Services definitions at this level have higher precedence than the Service definition at the **spec.service** level.                                                                                                                                                                                                                         |                                                                                                                                                                                                                                  |

> | **rules.path**                                   |  **YES**  | Specifies the path of the exposed Service. If the configured path of a rule overlaps with the path of another rule, e.g. if a rule defines * in the path, the configuration of both rules is applied.                                                                                                                                        | Value can be either exact path or a wildcard `*`.                                                                                                                                                                                |

Reviewer 3: ```suggestion
| **rules.path**                                   |  **YES**  | Specifies the path of the exposed Service. If the path specified in an Access Rule overlaps with the path of another Access Rule, both Access Rules are applied. This happens, for example, if one of the Access Rules' configurations contains `*`.                                                                                                             | The value can be either an exact path or a wildcard `*`.                                                                                                                                                                                 |
```

| **rules.methods**                                |  **NO**   | Specifies the list of HTTP request methods available for **spec.rules.path**. The list of supported methods is defined in [RFC 9910: HTTP Semantics](https://www.rfc-editor.org/rfc/rfc9110.html) and [RFC 5789: PATCH Method for HTTP](https://www.rfc-editor.org/rfc/rfc5789.html).                                                        |                                                                                                                                                                                                                                  |

> | **rules.noAuth**                                 |  **NO**   | Setting `noAuth` to `true` disables authorization.                                                                                                                                                                                                                                                                                           | When `noAuth` is set to true, it is not allowed to define `jwt` or `extAuth` on the same path.                                                                                                                                   |

Reviewer 3: ```suggestion
| **rules.noAuth**                                 |  **NO**   | Setting `noAuth` to `true` disables authorization.                                                                                                                                                                                                                                                                                           | When `noAuth` is set to true, it is not allowed to define the `jwt` or `extAuth` access strategy on the same path.                                                                                                                                   |
```

> | **rules.extAuths**                               |  **NO**   | Specifies the list of external authorizers. For more information see below and the [External Authorizer ADR](https://github.com/kyma-project/api-gateway/issues/938).                                                                                                                                                                        |                                                                                                                                                                                                                                  |

Reviewer 3: ```suggestion
| **rules.extAuths**                               |  **NO**   | Specifies the list of external authorizers. For more information see the [External Authorizer ADR](https://github.com/kyma-project/api-gateway/issues/938).                                                                                                                                                                        |                                                                                                                                                                                                                                  |
```

| **rules.extAuth.name**                           |  **NO**   | Specifies the name of the external authorizer.                                                                                                                                                                                                                                                                                               |                                                                                                                                                                                                                                  |

> | **rules.jwt**                                    |  **NO**   | Specifies the Istio JWT access strategy. For more information see [JWT Access Strategy](https://github.com/kyma-project/api-gateway/blob/main/docs/user/custom-resources/apirule/04-20-apirule-istio-jwt-access-strategy.md) and the bellow table.                                                                                           |                                                                                                                                                                                                                                  |

Reviewer 3: ```suggestion
| **rules.jwt**                                    |  **NO**   | Specifies the Istio JWT access strategy. For more information see [JWT Access Strategy](https://github.com/kyma-project/api-gateway/blob/main/docs/user/custom-resources/apirule/04-20-apirule-istio-jwt-access-strategy.md).                                                                                           |                                                                                                                                                                                                                                  |
```

| **rules.jwt.authentications.fromHeaders.name**   |  **YES**  | Specifies the name of the header.                                                                                                                                                                                                                                                                                                            |                                                                                                                                                                                                                                  |

> | **rules.jwt.authentications.fromHeaders.prefix** |  **NO**   | Specifies the prefix used before the JWT token. The default is `Bearer `.                                                                                                                                                                                                                                                                    |                                                                                                                                                                                                                                  |

Reviewer 3: ```suggestion
| **rules.jwt.authentications.fromHeaders.prefix** |  **NO**   | Specifies the prefix used before the JWT token. The default is `Bearer`.                                                                                                                                                                                                                                                                    |                                                                                                                                                                                                                                  |
```

| **rules.timeout**                                |  **NO**   | Specifies the timeout, in seconds, for HTTP requests made to **spec.rules.path**. Timeout definitions set at this level take precedence over any timeout defined at the **spec.timeout** level.                                                                                                                                              | The maximum timeout is limited to 3900 seconds (65 minutes).                                                                                                                                                                     |

> [!WARNING]

> > Ory handler is not supported in the v1beta2 version of the APIRule.

Reviewer 3: ```suggestion
> The Ory handler is not supported in version `v1beta2 of the APIRule.
```
